catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/operation/impl/QueryResponseImpl.java

```diff
@@ -302,7 +302,7 @@ private Result handleTake() {
         Result result = null;
         try {
             result = queue.take();
-            if (result == poisonPillResult) {
+            if (result == POISON_PILL_RESULT) {
```

Comment:
We should probably use object equality and make the field just `private final`. I think that having static in there makes it extraordinarily dangerous if someone doesn't do what you're expecting them to do with the query response. We would also need to override the equals method so that we can be check for some flag that tells us that is a poison pill result.